pandas-stubs/core/indexes/interval.pyi

```diff
@@ -11,80 +13,210 @@ import numpy as np
 import pandas as pd
 from pandas import Index
 from pandas.core.indexes.extension import ExtensionIndex
-from typing_extensions import TypeAlias
+from pandas.core.series import (
+    TimedeltaSeries,
+    TimestampSeries,
+)
+from typing_extensions import (
+    Never,
+    TypeAlias,
+)
 
 from pandas._libs.interval import (
     Interval as Interval,
-    IntervalMixin as IntervalMixin,
+    IntervalMixin,
 )
-from pandas._libs.tslibs.offsets import DateOffset
+from pandas._libs.tslibs.offsets import BaseOffset
 from pandas._typing import (
     DatetimeLike,
     DtypeArg,
     FillnaOptions,
     IntervalClosedType,
+    IntervalT,
     Label,
+    np_ndarray_bool,
     npt,
 )
 
 from pandas.core.dtypes.dtypes import IntervalDtype as IntervalDtype
 from pandas.core.dtypes.generic import ABCSeries
 
-_Edges: TypeAlias = Union[
+_EdgesInt: TypeAlias = Union[
     Sequence[int],
+    npt.NDArray[np.int64],
+    npt.NDArray[np.int32],
+    npt.NDArray[np.intp],
+    pd.Series[int],
+    pd.Int64Index,
+]
+_EdgesFloat: TypeAlias = Union[
     Sequence[float],
+    npt.NDArray[np.float64],
+    pd.Series[float],
+    pd.Float64Index,
+]
```

Comment:
I think the use of `pd.Int64Index` and `pd.Float64Index` here will cause issues with the nightly builds once #436 is merged in, so we'll have to figure out how to deal with that when in a `Union` like this.